### PR TITLE
Redis changes and application.ts refactoring

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/application.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  console.log(ERROR_NODE_CONFIG_ENV)
+}
+
+/**
+ * TODO: Implement a logger on the Node layer.
+ */
+console.log(environmentCheckText())
+
+// TODO: Testing that we can get the environment variables on AAT from the .yaml file
+console.log('ENV PRINT')
```
Comment (ficklephil) [inline api/application.ts]: I added these, but once we get up to production, when AKS is complete we'll have to remove these.

Comment (udaydenduluri33) [reply api/application.ts]: @ficklephil  I will remove them after it goes through AAT

File: src/environments/environment.prod.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export const environment = {
       idamLoginUrl: 'https://hmcts-access.service.gov.uk',
       idamWeb: 'https://hmcts-access.service.gov.uk',
       indexUrl: '/',
-      oauthCallbackUrl: 'oauth2/callback'
+      oauthCallbackUrl: '/oauth2/callback'
```
Comment (ficklephil) [inline src/environments/environment.prod.ts]: Just be careful here, I'm guessing it needs an extra slash now.

Comment (udaydenduluri33) [reply src/environments/environment.prod.ts]: I am not sure about the extra slash. Let's keep it for now. We had to remove it when were doing the openId connect @ficklephil

File: charts/xui-ao-webapp/values.yaml [hunk 3]
```diff
@@ -39,6 +49,12 @@ nodejs:
...
     FEATURE_HELMET_ENABLED: true
+    FEATURE_REDIS_ENABLED: true
+
+    # Redis
+    REDIS_KEY_PREFIX: 'activity:'
+    REDIS_TTL: 6000
+
   keyVaults:
     rpx:
       resourceGroup: rpx
```
Comment (paddybasi) [inline charts/xui-ao-webapp/values.yaml]: you will also need ao-webapp-redis-connection-string in the keyvaults secret

File: config/default.json [hunk 4]
```diff
@@ -79,6 +80,13 @@
     "hidePoweredBy": {
       "setTo": "XUI Server 1.0"
     }
+  },
+  "redis": {
+    "host": "xui-ao-webapp-prod.redis.cache.windows.net",
```
Comment (paddybasi) [inline config/default.json]: I have removed these as it should be coming from vault directly

File: api/routes.ts [hunk 5]
```diff
@@ -5,17 +5,15 @@ import healthCheck from './healthCheck'
...
 import stateRouter from './states'
+import pbaRouter from './updatePba'
 
 const router = express.Router({ mergeParams: true })
 // open routes
 router.use('/environment', environment)
 
 router.use(auth.attach)
 
-router.use('/logout', auth.logout)
```
Comment (paddybasi) [inline api/routes.ts]: bit confused about the removal of this?

Comment (udaydenduluri33) [reply api/routes.ts]: Its in the application.ts already. Hence, removed it

File: charts/xui-ao-webapp/values.yaml (hunk 3 above)
Comment (udaydenduluri33) [reply charts/xui-ao-webapp/values.yaml]: Added it now

File: config/default.json (hunk 4 above)
Comment (paddybasi) [reply config/default.json]: Also we use the connection string now and not an object key/value pair

Comment (udaydenduluri33) [reply config/default.json]: Done

**Pull request merged**